src/migrations/20220825000000-generic-approval-system.js

```diff
@@ -0,0 +1,324 @@
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      const ENTITY_TYPES = {
+        REPORT: 'report',
+        REPORTGOAL: 'report_goal',
+        REPORTOBJECTIVE: 'report_objective',
+        GOAL: 'goal',
+        GOALTEMPLATE: 'goal_template',
+        OBJECTIVE: 'objective',
+        OBJECTIVETEMPLATE: 'objectiveTemplate',
+      };
+
+      const APPROVAL_RATIO = {
+        ANY: 'any',
+        MAJORITY: 'majority',
+        TWOTHIRDS: 'two_thirds',
+        ALL: 'all',
+      };
+
+      const REPORT_STATUSES = {
+        DRAFT: 'draft',
+        DELETED: 'deleted',
+        SUBMITTED: 'submitted',
+        APPROVED: 'approved',
+        NEEDS_ACTION: 'needs_action',
+      };
+
+      try {
+        const loggedUser = '0';
+        // const transactionId = '';
+        const sessionSig = __filename;
+        const auditDescriptor = 'RUN MIGRATIONS';
+        await queryInterface.sequelize.query(
+          `SELECT
+            set_config('audit.loggedUser', '${loggedUser}', TRUE) as "loggedUser",
+            set_config('audit.transactionId', NULL, TRUE) as "transactionId",
+            set_config('audit.sessionSig', '${sessionSig}', TRUE) as "sessionSig",
+            set_config('audit.auditDescriptor', '${auditDescriptor}', TRUE) as "auditDescriptor";`,
+          { transaction },
+        );
+      } catch (err) {
+        console.error(err); // eslint-disable-line no-console
+        throw (err);
+      }
+
+      try {
+        await queryInterface.createTable('Approvals', {
+          id: {
+            allowNull: false,
+            autoIncrement: true,
+            primaryKey: true,
+            type: Sequelize.INTEGER,
+          },
+          entityType: {
+            allowNull: false,
+            default: null,
+            type: Sequelize.DataTypes.ENUM(
+              Object.keys(ENTITY_TYPES).map((k) => ENTITY_TYPES[k]),
```

Comment:
FYI, there is a built in method for doing this, IE 
```js 
Object.keys(ENTITY_TYPES).map((k) => ENTITY_TYPES[k])
``` 
is functionally the same as 
```js
Object.values(ENTITY_TYPES)
```
only with one less iteration